ghcide/src/Development/IDE/Plugin/Completions.hs

```diff
@@ -116,6 +117,12 @@ instance Hashable NonLocalCompletions
 instance NFData   NonLocalCompletions
 instance Binary   NonLocalCompletions
 
+-- | 40 may seem conservative but note that most editors limit how many completions
+--   are displayed in the screen, and most users rarely scroll.
+--   For instance, VSCode only shows 12 completions in its popup, and Emacs has a similar limit.
+maxCompletions :: Int
+maxCompletions = 40
+
```

Comment:
How about moving this into the setting, with the default value of, say, `40` (or anything prefered), so that one can configure it based on their machine power?

Reply:
Wish granted! I've moved it into config